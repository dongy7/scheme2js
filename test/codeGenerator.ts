import Compiler from '../src/compiler'

it('Generates function call', () => {
  const tokens = Compiler.tokenizer('(add 1 (sub 2 3))')
  const ast = Compiler.parser(tokens)
  const newAst = Compiler.transformer(ast)
  expect(Compiler.codeGenerator(newAst)).toMatchSnapshot()
})
