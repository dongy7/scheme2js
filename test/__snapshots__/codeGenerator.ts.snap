// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generates function call 1`] = `"add(1, sub(2, 3))"`;
